### PR TITLE
feat: redesign list handling in protographic

### PR DESCRIPTION
**Pull request by Noroth**
## Summary by CodeRabbit

* **New Features**
  * Improved support for GraphQL list types in Protocol Buffer conversion, introducing explicit wrapper messages for nullable and nested lists to enhance clarity and correctness.

* **Bug Fixes**
  * Enhanced proto generation accuracy for complex list scenarios, including nested and nullable lists.

* **Tests**
  * Added comprehensive tests covering diverse GraphQL list cases, including deep nesting and nullability variations.
  * Updated test snapshots to adopt the new wrapper message pattern for repeated fields.
  * Introduced tests ensuring consistent field numbering for auto-generated list wrapper types and their preservation across schema changes.

* **Chores**
  * Included the protographic module in the workspace configuration.
  * Expanded documentation with detailed explanations of list type handling and protobuf mapping conventions.

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: protographic/src/sdl-to-proto-visitor.ts [hunk 1]
```diff
@@ -1413,117 +1417,190 @@ Example:
...
+   * - [[String!]!]! → ListOfListOfString field_name = 1; (with nested wrapper messages)
+   * - [[String]] → ListOfListOfString field_name = 1; (with nested wrapper messages)
+   *
+   * @param type - The GraphQL list type to convert
+   * @returns ProtoType object containing the type name and whether it should be repeated
+   */
+  private handleListType(type: GraphQLList<GraphQLType>): ProtoType {
+    const listType = (isNonNullType(type) ? type.ofType : type) as GraphQLList<GraphQLType>;
+    const isNullableList = !isNonNullType(type);
+    const isNestedList = this.isNestedListType(listType);
```
Comment (endigma) [inline protographic/src/sdl-to-proto-visitor.ts]: I think this flattening of `GraphQLList<GraphQLType> | GraphQLNonNull<GraphQLList<GraphQLType>>` should happen in a helper function or at the call site, there are assertions both here and at call site that make the exact type of `type` unclear. I would also rename `type` to `graphqlType` like the calling function to avoid shadowing the type keyword

**Review by endigma — CHANGES_REQUESTED** [review]

**Review by endigma — APPROVED** [review]

File: protographic/tests/sdl-to-proto/11-lists.test.ts [hunk 2]
```diff
@@ -0,0 +1,1222 @@
...
+      message Item {
+        oneof value {
+        User user = 1;
+        Task task = 2;
+        }
+      }"
+    `);
+  });
+
+  it('should correctly generate protobuf for edge cases with empty lists and optional nesting', () => {
```
Comment (StarpTech) [inline protographic/tests/sdl-to-proto/11-lists.test.ts:1115]: Do we have tests that cover recursion and input types?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: protographic/src/sdl-to-proto-visitor.ts
Comment (StarpTech) [inline protographic/src/sdl-to-proto-visitor.ts:1]: Please add tests for the lock file

File: protographic/tests/sdl-to-proto/11-lists.test.ts (hunk 2 above)
Comment (Noroth) [reply protographic/tests/sdl-to-proto/11-lists.test.ts:1115]: done

File: protographic/src/sdl-to-proto-visitor.ts
Comment (Noroth) [reply protographic/src/sdl-to-proto-visitor.ts:1]: done
